src/Symfony/Component/Messenger/Transport/TransportFactoryInterface.php

```diff
@@ -20,7 +22,10 @@
  */
 interface TransportFactoryInterface
 {
-    public function createTransport(string $dsn, array $options): TransportInterface;
+    /**
+     * Creates a new transport using the dsn, options & serializer.
+     */
+    public function createTransport(string $dsn, array $options, SerializerInterface $serializer): TransportInterface;
```

Comment:
We decided that the serializer for a transport is probably an important enough concept that adding it as an argument (versus try to read it out of an option or the DSN) is probably best.